tessera.lic

```diff
@@ -0,0 +1,74 @@
+=begin
+  Documentation:  Tessera script based upon concept around Almanac.lic.
+=end
+
+custom_require.call(%w[common common-items drinfomon])
+
+class Tessera
+
+  def initialize
+    settings = get_settings
+    tessera_retry_interval = settings.tessera_retry_interval
+    startup_delay = settings.tessera_startup_delay
+
+    UserVars.tessera_last_use ||= Time.now - tessera_retry_interval
+    @no_use_scripts = settings.tessera_no_use_scripts 
+    @tessera = settings.tessera_noun
+
+    DRC.message("Tessera Startup Options:")
+    DRC.message("[tessera_noun: #{@tessera}]\n[tessera_startup_delay: #{startup_delay} (seconds)]\n[tessera_retry_interval: #{tessera_retry_interval} (seconds)]\n[tessera last attempt: #{UserVars.tessera_last_use}]\n[tessera_no_use_scripts: #{@no_use_scripts}]")
+
+    pause startup_delay
+    passive_loop
+  end
+
+  def use_tessera
+    settings = get_settings
+    tessera_retry_interval = settings.tessera_retry_interval
+
+    return if Time.now - UserVars.tessera_last_use < tessera_retry_interval
+    return if @no_use_scripts.any? { |name| Script.running?(name) }
+    return if XMLData.room_title.include? 'Carousel Chamber'
+    return if XMLData.room_title.include? 'Carousel Booth'
+    return if XMLData.room_title.include? 'Family Vault'
+    return if hidden?
+    return if invisible?
+    return if checkleft
+
+    pause 1 until DRC.pause_all
+    waitrt?
+    DRC.bput('retreat', 'You get', 'You retreat', 'You are already')
```

Comment:
`DRC.retreat` would be good here

Reply:
Thanks for the suggestion about DRC.retreat, if there's some standardized way to handle retreats it should be used.